plugins/amazonq/shared/jetbrains-community/src/software/aws/toolkits/jetbrains/services/amazonq/lsp/flareChat/ChatEditorUtils.kt

```diff
@@ -3,13 +3,42 @@
 
 package software.aws.toolkits.jetbrains.services.amazonq.lsp.flareChat
 
+import com.intellij.openapi.application.runReadAction
 import com.intellij.openapi.fileEditor.FileEditorManager
 import com.intellij.openapi.project.Project
+import org.eclipse.lsp4j.Position
+import org.eclipse.lsp4j.Range
 import org.eclipse.lsp4j.TextDocumentIdentifier
+import software.aws.toolkits.jetbrains.services.amazonq.lsp.model.aws.chat.CursorState
 import kotlin.io.path.Path
 
-fun getTextDocumentIdentifier(project: Project): TextDocumentIdentifier {
-    val selectedEditor = FileEditorManager.getInstance(project).selectedEditor
-    val filePath = Path(selectedEditor?.file?.path.orEmpty()).toUri()
+fun getTextDocumentIdentifier(project: Project): TextDocumentIdentifier? {
+    val selectedEditor = FileEditorManager.getInstance(project).selectedEditor ?: return null
+    val filePath = Path(selectedEditor.file.path).toUri()
     return TextDocumentIdentifier(filePath.toString())
 }
+
+fun getCursorState(project: Project): CursorState? {
+    return runReadAction {
+        val editor = FileEditorManager.getInstance(project).selectedTextEditor ?: return@runReadAction null
```

Comment:
let's make this a util because i think we might need a slightly different implementation for remote